go-chaos/internal/dataloss_sim_helper.go

```diff
@@ -0,0 +1,174 @@
+// Copyright 2022 Camunda Services GmbH
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package internal
+
+import (
+	"context"
+	"fmt"
+	corev1 "k8s.io/api/core/v1"
+	k8sErrors "k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/types"
+	"strconv"
+)
+
+const configMapName = "zeebe-control-pod-restart-flags"
+
+/*
+	Used for dataloss simulation test, to restrict when a deleted zeebe broker is restarted.
+
+This add an InitContainer to zeebe pods. The init container is blocked in an infinite loop, until the value  of `block_{node_id}` in the config map is set to false.
+To restrict when a deleted pod is restarted, first update the configmap and set the respective `block_{node_id}` true.
+Then delete the pod. Once it is time to restart the pod, update the config map to set the `block_{nodeId}` to false.
+The updated config map will be eventually (usually with in a minute) by the init container and breaks out of the loop.
+The init container exits and the zeebe container will be started.
+*/
+func (c K8Client) ApplyInitContainerPatch() error {
+	// apply config map
+	err := createConfigMapForInitContainer(c)
+	if err != nil {
+		fmt.Printf("Failed to create config map %s", err)
+		return err
+	}
+
+	statefulSet, err := c.GetZeebeStatefulSet()
+	if err != nil {
+		fmt.Printf("Failed to get statefulset %s", err)
+		return err
+	}
+
+	// Adds init container patch
+	patch := []byte(`{
```

Comment:
:wrench: was thinking whether we could extract this into an own function since it is quite long